### PR TITLE
 Welcome defaultbackend to the ingress-gce repo

### PR DESCRIPTION
**Pull request by jonpulsifer**
This commit introduces the 404-server to the ingress-gce repository.

 Previously, it was held in the kubernetes/ingress-nginx repo, but the
 nginx ingress controller no longer requires it.

 Fixes https://github.com/kubernetes/ingress-gce/issues/498

Signed-off-by: Jonathan Pulsifer <jonathan.pulsifer@shopify.com>

### CODE REVIEW

**Comment by aledbf** [comment]
@bowei friendly ping

**Comment by bowei** [comment]
@jonpulsifer -- is it possible to split into two commits:

1) with all the changes needs to wire in the new code
2) just the code copied over

It's hard to review in the github UI otherwise

**Comment by jonpulsifer** [comment]
Pardon the delay 😄 

I think this is ok now? A few things:

1. 404-server now builds like the other binaries
2. metrics have been removed (byebye vendor/)
3. works on my machine 😂 

The last nit, I think, is that the image builds with the following image name `gcr.io/google_containers/ingress-gce-404-server-amd64`

I assume we would like it to maintain its original name `defaultbackend` and version but I don't know how you'd like to do this.

Ref https://console.cloud.google.com/gcr/images/google_containers/GLOBAL/defaultbackend

File: cmd/404-server/OWNERS [hunk 1]
```diff
@@ -0,0 +1,12 @@
+approvers:
```
Comment (rramkumar1) [inline cmd/404-server/OWNERS]: We don't need a separate OWNERS for this. The top level OWNERS should suffice.

**Comment by rramkumar1** [comment]
@jonpulsifer Can you also add the binary name to the Makefile?

**Comment by jonpulsifer** [comment]
The binary was added in https://github.com/kubernetes/ingress-gce/pull/503/commits/4bb3afe820d0a1e82c2ea710c1dc5fc93b28514b and the owners have been removed

**Comment by rramkumar1** [comment]
@jonpulsifer Thanks alot, this looks great!

/lgtm
/hold

Will get back to you today regarding naming of the image.

**Comment by rramkumar1** [comment]
@jonpulsifer Naming is fine. I'll take care of updating any stale references

/hold remove

**Pull request merged**